plugins/source/azure/resources/services/monitor/diagnostic_settings.go

```diff
@@ -1,281 +1,130 @@
 			},
 		},
 	}
 }
 
-// ====================================================================================================================
-//
-//	Table Resolver Functions
-//
-// ====================================================================================================================
-func fetchMonitorDiagnosticSettings(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan<- interface{}) error {
-	cl := meta.(*client.Client)
-	resSvc := cl.Services().Resources.Resources
-	monSvc := cl.Services().Monitor.DiagnosticSettings
-	resResponse, err := resSvc.List(ctx, "", "", nil)
-	if err != nil {
-		return diag.WrapError(err)
-	}
-	rs := resResponse.Values()
-	ids := make([]string, 0, len(rs))
-	ids = append(ids, "/subscriptions/"+cl.SubscriptionId)
-	for _, r := range rs {
-		ids = append(ids, *r.ID)
+func isResourceTypeNotSupported(err error) bool {
+	var azureErr *azure.RequestError
+	if errors.As(err, &azureErr) {
+		return azureErr.ServiceError != nil && azureErr.ServiceError.Code == "ResourceTypeNotSupported"
 	}
+	return false
+}
 
-	g, _ := errgroup.WithContext(ctx)
-	g.SetLimit(maxGoroutines)
-	for _, i := range ids {
-		id := i
-		g.Go(func() error {
-			response, err := monSvc.List(ctx, id)
-			if err != nil {
-				if isResourceTypeNotSupported(err) {
-					return nil
-				}
-				return diag.WrapError(err)
-			}
-			if response.Value == nil {
-				return nil
-			}
-			for _, v := range *response.Value {
-				res <- diagnosticSetting{
-					DiagnosticSettings: v.DiagnosticSettings,
-					ID:                 v.ID,
-					Name:               v.Name,
-					Type:               v.Type,
-					ResourceURI:        id,
-				}
-			}
-			return nil
-		})
-	}
+// diagnosticSettingResource is a custom copy of insights.DiagnosticSettingsResource with extra ResourceURI field
+type diagnosticSettingResource struct {
+	insights.DiagnosticSettingsResource
+	ResourceURI string
+}
 
-	err = g.Wait()
+func fetchMonitorDiagnosticSettings(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan<- interface{}) error {
+	svc := meta.(*client.Client).Services().Monitor.DiagnosticSettings
 
-	return diag.WrapError(err)
-}
-func fetchMonitorDiagnosticSettingMetrics(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan<- interface{}) error {
-	p := parent.Item.(diagnosticSetting)
-	if p.DiagnosticSettings == nil ||
-		p.DiagnosticSettings.Metrics == nil {
-		return nil
+	resource := parent.Item.(resources.GenericResourceExpanded)
+	response, err := svc.List(ctx, *resource.ID)
+	if err != nil {
+		if isResourceTypeNotSupported(err) {
+			return nil
+		}
+		return errors.WithStack(err)
 	}
-
-	res <- *p.DiagnosticSettings.Metrics
-	return nil
-}
-func fetchMonitorDiagnosticSettingLogs(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan<- interface{}) error {
-	p := parent.Item.(diagnosticSetting)
-	if p.DiagnosticSettings == nil ||
```

Comment:
`DiagnosticSettings` is an embedded struct so we can access `Logs` directly (e.g. `p.Logs`).
Anyway I removed this extra relation and made `Logs` a JSON column